inc/endpoints/class-components-endpoint.php

```diff
@@ -131,14 +131,23 @@ public function register_rest_routes() {
 	 * @return array
 	 */
 	public function get_route_response( $request ) {
+
+		// phpcs:ignore WordPress.WP.GlobalVariablesOverride.OverrideProhibited
+		global $wp;
+
 		/**
 		 * Action fired on the request.
 		 *
 		 * @param \WP_REST_Request $request  WP_REST_Request object.
 		 */
 		do_action( 'wp_irving_components_request', $request );
 
-		$this->params = $request->get_params();
+		// Remove any request parameters that haven't been allow-listed by the
+		// global $wp object's $public_query_vars array.
+		$this->params = array_intersect_key(
+			$request->get_params(),
+			array_flip( $wp->public_query_vars )
```

Comment:
oh that is clever

Reply:
Here's the full list of where WP adds public query vars:
https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/class-wp.php#L9-L17

Are we sure we want/need to support all of these explicitly? Many of them should be handled by the route parsing on the WP side, I would think.